### PR TITLE
Update test workflow

### LINKED ISSUE #201: Run JS tests on PRs
Currently, only the PHP tests are being run.

### PR DESCRIPTION
**Pull request by adamwoodnz**
Closes #201 

Add running JS tests after PHP
Change to only run on pushes to trunk, and any PR

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,6 +1,10 @@
 name: Test
 
-on: [push, pull_request]
```
Comment (adamwoodnz) [inline .github/workflows/test.yml:3]: This is currently causing a duplicate run on every PR, eg.

Screen Shot 2023-05-31 at 12 39 26 PM

Comment (StevenDufresne) [reply .github/workflows/test.yml:3]: Use this format:
```
    push:
        branches: [trunk]
    pull_request:
```
See: https://github.com/WordPress/wporg-main-2022/blob/trunk/.github/workflows/linters.yml

Comment (adamwoodnz) [reply .github/workflows/test.yml:3]: Yep, see change below 😄 

This PR fixes the issue, sorry for confusion

**Review by outdoor2kode — APPROVED** [review]
Nice finding and thanks for the fix. LGTM 👍

**Review by outdoor2kode — APPROVED** [review]
Nice finding and thanks for the fix. LGTM 👍

**Pull request merged**